pkg/signatures/sms.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signatures
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/google/exposure-notifications-verification-server/internal/project"
+)
+
+const (
+	// authPrefix is the string to prepend to the authentication
+	// signature.
+	authPrefix = "Authentication: "
+
+	// Dot represents a period, which is used as a separator in some signatures.
+	dot = "."
+)
+
+// SMSPurpose is an SMS purpose, used in signature calculation.
+type SMSPurpose string
+
+const (
+	// SMSPurposeENReport is an SMS purpose for EN reporting.
+	SMSPurposeENReport SMSPurpose = "EN Report"
+)
+
+// SMSSignature returns the signature of the message uses the provided signer.
+func SMSSignature(signer crypto.Signer, keyID string, t time.Time, purpose SMSPurpose, phone, body string) (string, error) {
+	t = t.UTC()
+	signingString := smsSignatureString(t, purpose, phone, body)
```

Comment:
as above - I Think we want to add + " Authentication:" to the end of the body here (well, probably in the smsSignatureString message itself)

Reply:
@mikehelmick or do we expect the SMS template itself to declare it like `blah blah Authorization:[[auth]]`?

Reply:
no - I think it needs to be inserted here. Because using SMS auth will be optional. If enabled, we should insert it.

Reply:
Then this function needs to return a _new_ SMS body, which feels like a really weird API.

Reply:
no, it can still return what should be appended to the body

Reply:
Okay, done